saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.saga.engine.config;
+
+import io.seata.saga.engine.impl.DefaultStateMachineConfig;
+import io.seata.saga.engine.store.db.DbStateLangStore;
+import io.seata.saga.engine.store.db.DbAndReportTcStateLogStore;
+import io.seata.saga.tm.DefaultSagaTransactionalTemplate;
+import io.seata.saga.tm.SagaTransactionalTemplate;
+import org.springframework.beans.factory.DisposableBean;
+
+import javax.sql.DataSource;
+
+/**
+ * DbStateMachineConfig
+ *
+ * @author lorne.cl
+ */
+public class DbStateMachineConfig extends DefaultStateMachineConfig implements DisposableBean {
+
+    private DataSource                dataSource;
+    private String                    applicationId;
+    private String                    txServiceGroup;
+    private String                    tablePrefix                  = "seata_";
+    private String                    dbType                       = "mysql";
```

Comment:
The dbType property is designed for the future. DML has no special statements. In theory, it can support all databases. Currently tested mysql, db2, oracle, h2. We provide mysql, db2, oracle, h2 build scripts by default.